bitcoin/src/util/hash.rs

```diff
@@ -39,6 +39,44 @@ where
 
 /// Calculates the merkle root of an iterator of *hashes*.
 ///
+/// This fcuntion calculate the root from a merkle path when we need to derive the root from the
+/// txs list [bitcoin_merkle_root] should be used.
+///
+/// # Returns
+/// - `None` if `hashes` is empty. The merkle root of an empty tree of hashes is undefined.
+/// - `Some(hash)` if `hashes` contains one element. A single hash is by definition the merkle root.
+/// - `Some(merkle_root)` if length of `hashes` is greater than one.
+pub fn bitcoin_merkle_root_from_path<T, I>(mut hashes: I) -> Option<T>
+where
+    T: Hash + Encodable,
+    <T as Hash>::Engine: io::Write,
+    I: Iterator<Item = T>,
+{
+    let first = hashes.next()?;
+    let second = match hashes.next() {
+        Some(second) => second,
+        None => return Some(first),
+    };
+
+    let mut encoder = T::engine();
+    let mut root = first;
```

Comment:
It'd be nicer to write `let mut root = T::from_engine(encoder);` below and remove this line.

Reply:
👍